services/slam/builtin/builtin.go

```diff
@@ -457,6 +480,41 @@ func (slamSvc *builtIn) GetMap(
 	if err != nil {
 		return "", nil, nil, err
 	}
+	reqPCMap := &pb.GetPointCloudMapRequest{
+		Name: name,
+	}
+
+	var imData image.Image
+	var vObj *vision.Object
+
+	if slamSvc.dev {
+		slamSvc.logger.Debug("IN DEV MODE (map request)")
+
+		resp, err := slamSvc.clientAlgo.GetPointCloudMap(ctx, reqPCMap)
+
+		if err != nil {
+			return "", imData, vObj, errors.Errorf("error getting SLAM map (%v) : %v", mimeType, err)
+		}
+		pointcloudData := resp.GetPointCloudPcd()
+		if pointcloudData == nil {
+			return "", nil, nil, errors.New("get map read pointcloud unavailable")
+		}
+		pc, err := pc.ReadPCD(bytes.NewReader(pointcloudData))
+		if err != nil {
+			return "", nil, nil, errors.Wrap(err, "get map read pointcloud failed")
+		}
+
+		vObj, err = vision.NewObject(pc)
+		if err != nil {
+			return "", nil, nil, errors.Wrap(err, "get map creating vision object failed")
+		}
+
+		return rdkutils.MimeTypePCD, imData, vObj, nil
```

Comment:
[q] Does this break frontend if someone requests a jpeg map while using the dev flag? Or does frontend already check the return mimeType

Reply:
Good question! I had ignored this since mimeType will not be a field in `GetPointCloudMap` but you are right that is worth checking here